Corona-Warn-App/src/main/java/de/rki/coronawarnapp/ui/submission/qrcode/scan/SubmissionQRCodeScanFragment.kt

```diff
@@ -86,16 +87,57 @@ class SubmissionQRCodeScanFragment :
                 else -> View.GONE
             }
             if (ApiRequestState.SUCCESS == state.apiRequestState) {
-                if (state.testResult == CoronaTestResult.PCR_POSITIVE) {
-                    doNavigate(
-                        SubmissionQRCodeScanFragmentDirections
-                            .actionSubmissionQRCodeScanFragmentToSubmissionTestResultAvailableFragment()
-                    )
-                } else {
-                    doNavigate(
-                        SubmissionQRCodeScanFragmentDirections
-                            .actionSubmissionQRCodeScanFragmentToSubmissionTestResultPendingFragment()
-                    )
+                when (state.testResult) {
+                    CoronaTestResult.PCR_POSITIVE ->
+                        doNavigate(
+                            SubmissionQRCodeScanFragmentDirections
+                                .actionSubmissionQRCodeScanFragmentToSubmissionTestResultAvailableFragment(
+                                    CoronaTest.Type.PCR
+                                )
+                        )
+                    CoronaTestResult.PCR_OR_RAT_PENDING -> {
+                        if (state.testType == CoronaTest.Type.RAPID_ANTIGEN) {
+                            doNavigate(
+                                SubmissionQRCodeScanFragmentDirections
+                                    .actionSubmissionQRCodeScanFragmentToSubmissionTestResultPendingFragment(
+                                        testType = CoronaTest.Type.RAPID_ANTIGEN
+                                    )
+                            )
+                        } else {
+                            doNavigate(
+                                SubmissionQRCodeScanFragmentDirections
+                                    .actionSubmissionQRCodeScanFragmentToSubmissionTestResultPendingFragment(
+                                        testType = CoronaTest.Type.PCR
+                                    )
+                            )
+                        }
+                    }
+                    CoronaTestResult.PCR_NEGATIVE,
+                    CoronaTestResult.PCR_INVALID,
+                    CoronaTestResult.PCR_REDEEMED ->
+                        doNavigate(
+                            SubmissionQRCodeScanFragmentDirections
+                                .actionSubmissionQRCodeScanFragmentToSubmissionTestResultPendingFragment(
+                                    testType = CoronaTest.Type.RAPID_ANTIGEN
```

Comment:
I think this should be `PCR` ?  ,`when` cases says `PCR_`

Reply:
Yes, thanks, I'll update it.